### ISSUE TITLE
Custom parse format not working with ISO UTC date

### ISSUE THREAD

**Issue by kb3eua** [iamkun/dayjs#1498]
**Describe the bug**
Date String: `2021-05-12T18:51:55.434Z`
Parsing with
```js
dayjs('2021-05-12T18:51:55.434Z', 'YYYY-MM-DDTHH:mm:ss.SSS[Z]')
```
... results in the date being read in as local to my timezone, as `2021-05-12T18:51:55.434-0400` instead of as UTC.
Parsing with
```js
dayjs('2021-05-12T18:51:55.434Z', 'YYYY-MM-DDTHH:mm:ss.SSSZ')
```
... results in `Invalid Date`

**Expected behavior**
Works correctly when passing the format as an object (which I can't find any documentation on) instead of a string using either of these format strings.
```js
dayjs('2021-05-12T18:51:55.434Z', {format:'YYYY-MM-DDTHH:mm:ss.SSS[Z]'})
dayjs('2021-05-12T18:51:55.434Z', {format:'YYYY-MM-DDTHH:mm:ss.SSSZ'})
```
... both produce `2021-05-12T14:51:55.434-0400`, which is correctly adjusted for my timezone.

**Information**
 - Day.js Version v1.10.4
 - OS: Windows 10
 - Browser: Chrome 90
 - Time zone: GMT-04:00 EDT (Eastern Daylight Time)

**Comment by kb3eua** [comment]
Looks like there's a fix for this in the `dev` branch: #1359

When can we expect that to be released?

**Comment by ghiscoding** [comment]
@kb3eua that shipped few days ago in this [v1.10.5](https://github.com/iamkun/dayjs/releases/tag/v1.10.5) release, but it unfortunately doesn't work in strict mode as I wrote in a comment [here](https://github.com/iamkun/dayjs/issues/1358#issuecomment-774296660)
